### PR TITLE
[RSDK-8859][RSDK-8386] Remove raspberry pi built-in model, test workflows and ld flag for static library. 

### PR DESCRIPTION
**Pull request by randhid**
What it says on the tin.

Are there guides anywhere in our wiki for our app image build flows? I think some of the relevant libraries are used for packaging the app image, but I don't know how to remove them.

This is blocked from merging until the migrations are complete.
https://github.com/viamrobotics/app/pull/6275

### CODE REVIEW

File: .github/workflows/pullrequest-trusted.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ on:
 jobs:
   test:
     if: (github.event.label.name == 'safe to test' || github.event.label.name == 'appimage')
-    uses: viamrobotics/rdk/.github/workflows/test.yml@main
+    uses: ./.github/workflows/test.yml
```
Comment (Otterverse) [inline .github/workflows/pullrequest-trusted.yml:19]: This is just temporary, right?

**Review by Otterverse — APPROVED** [review]
Looks good to me (provided the one workflow tag gets reverted before merge)

To answer your question about appimages... nothing special for Pi there. Pigpio was being statically linked, so it's only the dev environment that contained it. We don't bundle or ship it.

If you're curious WHERE stuff was defined for appimages anyway though, it's in here: https://github.com/viamrobotics/rdk/blob/main/etc/packaging/appimages/viam-server-aarch64.yml (and the similar files for other architectures.)

File: .github/workflows/pullrequest-trusted.yml (hunk 1 above)
Comment (randhid) [reply .github/workflows/pullrequest-trusted.yml:19]: Yes, I was checking that the tests would pass on my branch first, this blocks it from getting merged anyway, right?

**Comment by randhid** [comment]
> If you're curious WHERE stuff was defined for appimages anyway though, it's in here: https://github.com/viamrobotics/rdk/blob/main/etc/packaging/appimages/viam-server-aarch64.yml (and the similar files for other architectures.)

This is where I was looking actually, I didn't know if merging what I have right now would be complete without removing the libraries in packaging, especially since these files are ignored, but if it is unnecessary to edit them, great.

**Review by abe-winter — APPROVED** [review]

**Review by susmitaSanyal — APPROVED** [review]
YEY!

**Pull request merged**